File: 08. Analisis Kompleksitas/C. Bilangan Agak Prima/c_bilangan_agak_prima.pas
program c_bilangan_agak_prima;
var
  N, i : integer;
  j, num, factor : longint;
begin
  readln(N);
  for i := 1 to N do begin
    readln(num);
    factor := 0;
    for j := 1 to num do begin
      if (num Mod j = 0) then
        factor := factor + 1;
      if (factor > 4) then break;
    end;
    if (factor <= 4) then
      writeln('YA')
    else
      writeln('BUKAN');
  end;
end.

File: 08. Analisis Kompleksitas/C. Bilangan Agak Prima/c_bilangan_agak_prima.cpp
#include <iostream>
int main(){
  int N, num, factor;
  std::cin >> N;
  for (int i = 0; i < N; i++){
    std::cin >> num;
    factor = 0;
    for (int j = 1; j <= num; j++){
      if (num % j == 0){
        factor++;
      }
      if (factor > 4){
        break;
      }
    }
    factor <= 4 ? std::cout << "YA" : std::cout << "BUKAN";
    std::cout << std::endl;
  }
}
